subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc

```diff
@@ -0,0 +1,849 @@
+
+Bootstrapping can either be used to create a file from the beginning, or also to _update_ an existing file with new information.
+Therefore, it's recommended to always use the same parameters once you started bootstrapping.
+
+The dependency verification file can be generated with the following CLI instructions:
+
+
+----
+gradle --write-verification-file sha256 help
+----
+
+The `write-verification-file` flag requires the list of <<#sec:checksum-verification,checksums>> that you want to generate or `pgp` for <<#sec:signature-verification,signatures>>.
+
+Executing this command line will cause Gradle to:
+
+* resolve all <<declaring_dependencies.adoc#sec:resolvable-consumable-configs,resolvable configurations>>, which includes:
+** configurations from the root project
+** configurations from all subprojects
+** configurations from `buildSrc`
+** included builds configurations
+** configurations used by plugins
+* download all artifacts discovered during resolution
+* compute the requested checksums and possibly verify signatures depending on what you asked
+* At the end of the build, generate the configuration file which will contain the inferred _verification metadata_
+
+As a consequence, the `verification-metadata.xml` file will be used in subsequent builds to verify dependencies.
+
+[WARNING]
+====
+There are dependencies that Gradle _cannot_ discover this way.
+In particular, you will notice that the CLI above uses the `help` task.
+If you don't specify any task, Gradle will automatically run the default task and generate a configuration file at the end of the build too.
+
+The difference is that Gradle _may_ discover more dependencies and artifacts depending on the tasks you execute.
+As a matter of fact, Gradle cannot automatically discover _detached configurations_, which are basically dependency graphs resolved as an internal implementation detail of the execution of a task: they are not, in particular, declared as an input of the task because they effectively depend on the configuration of the task at execution time.
+
+A good way to start is just to use the simplest task, `help`, which will discover as much as possible, and if subsequent builds fail with a verification error, you can re-execute generation with the appropriate tasks to "discover" more dependencies.
+====
+
+[[sec:verification-dry-mode]]
+=== Using dry mode
+
+There are situations where you would just want to _see_ what the generated verification metadata file would look like without actually changing the existing one or overwriting it.
+
+For this purpose, you can just add `--dry-run`:
+
+
+----
+gradle --write-verification-file sha256 help --dry-run
+----
+
+Then instead of generating the `verification-metadata.xml` file, a _new file_ will be generated, called `verification-metadata.dryrun.xml`.
```

Comment:
Technically speaking, everybody's using VCS, so generating files like `.dryrun.xml` is painful :(

It would be great if dry run overwrote the original file (then the user could use `git diff` to analyze the changes). If you think it is important to have a case when a file with another name is created, then `org.gradle.verfication...` property could be used to configure that.

Reply:
I think you misunderstood: by default, if you don't use `--dry-run`, it will _not_ create a new file but overwrite the existing one. Which is ok if you use VCS, you'll get the diff.

However, we _also_ provide a way to generate a file without overwriting, which is a convenience that I personally used several times when dogfooding.

Reply:
I see now. For some reason, I don't see a button to "close the discussion". Is it because I created "comments" rather than "started the review"?

Reply:
I'll close it. If you didn't understand it means I need to clarify :)